### PR TITLE
XIVY-14570 Upgrade to Maven 3.9

### PR DESCRIPTION
**Pull request by alexsuter**
I think it's time to move on.

image

Maven 3.9 is now also bundled as part of M2E. 

Should also fix security issues:
https://github.com/axonivy/project-build-plugin/security/dependabot

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -81,7 +81,7 @@
     <dependency>
       <groupId>org.apache.maven.shared</groupId>
       <artifactId>maven-filtering</artifactId>
-      <version>3.2.0</version>
+      <version>3.3.2</version>
```
Comment (ivy-rew) [inline pom.xml:84]: I think we had to downgrade this dependency multiple times; there was a problem only occurring the other day when re-integrating (e.g. in demo-projects) build.

**Review by ivy-rew — APPROVED** [review]
yes, 3.9 is perfect for the next release. :heavy_check_mark:

File: pom.xml (hunk 1 above)
Comment (alexsuter) [reply pom.xml:84]: Ok I will try it. Otherwise revert it.

Comment (alexsuter) [reply pom.xml:84]: Seems to work:
https://jenkins.ivyteam.io/job/demo-projects/job/master/

**Pull request merged**